RecoTracker/FinalTrackSelectors/python/MergeTrackCollections_cff.py

```diff
@@ -50,25 +54,23 @@
 
 from Configuration.Eras.Modifier_fastSim_cff import fastSim
 fastSim.toReplaceWith(generalTracksTask, 
-                      cms.Task(
-        duplicateTrackCandidates,
-        mergedDuplicateTracks,
-        duplicateTrackClassifier
-        )
+                      cms.Task(duplicateTrackCandidates,
+                               mergedDuplicateTracks,
+                               duplicateTrackClassifier)
 )
+
 def _fastSimGeneralTracks(process):
     from FastSimulation.Configuration.DigiAliases_cff import loadGeneralTracksAlias
     loadGeneralTracksAlias(process)
 modifyMergeTrackCollections_fastSimGeneralTracks = fastSim.makeProcessModifier( _fastSimGeneralTracks )
 
 import RecoTracker.FinalTrackSelectors.trackListMerger_cfi
 conversionStepTracks = RecoTracker.FinalTrackSelectors.trackListMerger_cfi.trackListMerger.clone(
-    TrackProducers = cms.VInputTag(cms.InputTag('convStepTracks')),
-    hasSelector=cms.vint32(1),
-    selectedTrackQuals = cms.VInputTag(cms.InputTag("convStepSelector","convStep")
-                                       ),
+    TrackProducers     = ['convStepTracks'],
+    hasSelector        = [1],
+    selectedTrackQuals = ['convStepSelector:convStep'],
     setsToMerge = cms.VPSet( cms.PSet( tLists=cms.vint32(1), pQual=cms.bool(True) )
                              ),
```

Comment:
```suggestion
    setsToMerge = [ cms.PSet( tLists=cms.vint32(1), pQual=cms.bool(True) )
                             ],
```
this still works.

I'd think that a nameless tuple like  could still work for PSet, like ```[ ( tLists=cms.vint32(1), pQual=cms.bool(True) ]```, but this is apparently not supported.
@makortel do you think it would be practical to add?

Reply:
I don't think `( tLists=cms.vint32(1), pQual=cms.bool(True)` would be legal syntax
```
>>> (foo=2)
  File "<stdin>", line 1
    (foo=2)
        ^
SyntaxError: invalid syntax
```
So using a keyword argument would require `dict(foo=2)`, which is already used for modifying an existing PSet.

Reply:
dict does not really apply here because full contents of ```setsToMerge``` are replaced (we are not modifying just one of the elements of this array.

Just to reiterate, my question was, if we have support for VInputTag to be initialized from ```[ "a", b", ("c", "d") ] ```, can we also get a VPSet to be initialized without listing the element type?

Reply:
With `VInputTag` a compact (maybe event too compact) is possible because the meaning of each element of the sequence of strings is known by its position in the sequence. In `PSet` the element keys are the parameter names, so `dict` would be a natural "shortcut". But we already use `dict` for something else, and I'd think anything along `("tlists", cms.vint32(1), "pQual", cms.bool(True))` would be (much) less clear than just using the `cms.PSet`.

Reply:
sorry, I meant 
instead of 
```cms.VPSet( cms.PSet( tLists=cms.vint32(1), pQual=cms.bool(True) ) )```
to be able to say
```[ ( tLists=cms.vint32(1), pQual=cms.bool(True) ) ]```
this is not serving the full goal of "drop type specs", but it would at least allow to insert elements to a container of a known type (vector of cms.PSet) without need to explicitly specify the type of the element (cms.PSet)

Reply:
The problem is that  `( tLists=cms.vint32(1), pQual=cms.bool(True) )` is not valid syntax.

Reply:
ah, so, that's python side, not something that's related with how cms.PSet can possibly be initialized

Reply:
Right. If anyone comes up with a viable syntax for a shortcut, I'm happy to revisit the question.

Reply:
how about ```[ { 'tLists' : cms.vint32(1), 'pQual' : cms.bool(True) } ]``` ?
it would be a valid syntax, but it comes out with ```raise TypeError("wrong type being extended to container "+self._labelIfAny())```

Reply:
The `{ 'tLists' : cms.vint32(1), 'pQual' : cms.bool(True) }` is a dictionary equivalent to `dict(tLists = cms.vint32(1), pQual = cms.bool(True))`.

Reply:
OK, syntactically , this ```dict(``` may be more transparent.
I guess from the above the point is that ```dict``` doesn't do the right thing here (yet?).

Reply:
> OK, syntactically , this `dict(` may be more transparent.
> I guess from the above the point is that `dict` doesn't do the right thing here (yet?).

OTOH, the difference to type `cms.PSet` vs `dict(` is rather minor; IIRC this is also the only vector type that needs this. So, we may just live with it for now.

Reply:
> OK, syntactically , this `dict(` may be more transparent.
> I guess from the above the point is that `dict` doesn't do the right thing here (yet?).

We already use `dict()` to modify an existing PSet. I think using it to denote new PSet in one context would be confusing.